examples/financialcircuit/testcases.go

```diff
@@ -0,0 +1,144 @@
+		Ticker: "ENB 5.375 27-Sep-2027",
+	}
+
+	// TODO - Cpt1 Quote is always the acepted quote, see how to change that
+	// 2nd parameter is always the accepted quote
+	toRet[0] = getQuotesValue(bond, "92.63", "92.63", "95", "Initiator Party selected Cpt1")                               // test case 1 - 2 quotes have same value.
+	toRet[1] = getQuotesValue(bond, "91.71", "91.71", "91.71", "Initiator Party selected Cpt1")                            // test case 2
+	toRet[2] = getQuotesValue(bond, "0", "0", "0", "Generate proof fails - all quotes are zero")                           // test case 10
+	toRet[3] = getQuotesValue(bond, "92.63", "92.63", "92.63", "Initiator Party selected Cpt1")                            // test case 2
+	toRet[4] = getQuotesValue(bond, "97.63", "94.63", "95.63", "Generate proof fails - Initiator selected a higher quote") // test case 8
+	toRet[5] = getQuotesValue(bond, "-97.63", "-94.63", "-95.63", "Generate proof fails - Negative quotes")                // test case 11
+
+	bond = &Bond{
+		Isin:   "CA29250NAT25",
+		Size:   "1550000",
+		Ticker: "ENB 1.375 10-Sep-2025",
+	}
+
+	toRet[6] = getQuotesValue(bond, "91.63", "92.63", "95.63", "Initiator Party selects the smallest quote")
+	toRet[7] = getQuotesValue(bond, "93", "98", "94", "Initiator Party selects the smallest integer quote")
+
+	//test cases for exotic bonds, callable bond, step up, FRN. need to update quotes
+	bond = &Bond{
+		Isin:   "6625HKC3",
+		Size:   "550000",
+		Ticker: "JPM 3.125% 01/23/2025 Callable",
+	}
+	toRet[8] = getQuotesValue(bond, "91.63", "92.63", "95.63", "Initiator Party selects the smallest quote")
+	toRet[9] = getQuotesValue(bond, "93", "98", "94", "Initiator Party selects the smallest integer quote")
+	// Callable with a step up on 6/23/26
+	bond = &Bond{
+		Isin:   "8128GT91",
+		Size:   "450000",
+		Ticker: "JPM  STEP 06/23/2030 Callable Step 06/23/2026 @ 2.25",
+	}
+	toRet[10] = getQuotesValue(bond, "91.63", "92.63", "95.63", "Initiator Party selects the smallest quote")
+	toRet[11] = getQuotesValue(bond, "93", "98", "94", "Initiator Party selects the smallest integer quote")
+	// FRN/Variable rate bond
+	bond = &Bond{
+		Isin:   "89114QCR7",
+		Size:   "600000",
+		Ticker: "The Toronto-Dominion VAR 03/04/2024",
+	}
+	toRet[12] = getQuotesValue(bond, "91.63", "92.63", "95.63", "Initiator Party selects the smallest quote")
+	toRet[13] = getQuotesValue(bond, "93", "98", "94", "Initiator Party selects the smallest integer quote")
+	bond = &Bond{
+		Isin:   "46625HKC3",
+		Size:   "625000",
+		Ticker: "JPM 3.125% 01/23/2025 Callable",
+	}
+	toRet[14] = getQuotesValue(bond, "91.63", "92.63", "95.63", "Initiator Party selects the smallest quote")
+	toRet[15] = getQuotesValue(bond, "93", "98", "94", "Initiator Party selects the smallest integer quote")
+
+	return toRet
+}
+
+func getQuotesValue(bond *Bond, quoteCpt1, quoteCpt2, quoteCpt3, message string) TestCase {
+
+	_quoteCpt1, err := decimal.NewFromString(quoteCpt1)
```

Comment:
err values on lines 96-97 are not checked

Reply:
good. I will review err values and checks